client/cody/src/extension.ts

```diff
@@ -1,15 +1,86 @@
 import * as vscode from 'vscode'
+import { LanguageClient, LanguageClientOptions, ServerOptions, TransportKind } from 'vscode-languageclient/node'
 
 import { PromptMixin, languagePromptMixin } from '@sourcegraph/cody-shared/src/prompt/prompt-mixin'
 
 import { CommandsProvider } from './command/CommandsProvider'
+import { getConfiguration } from './configuration'
 import { ExtensionApi } from './extension-api'
+import { CODY_ACCESS_TOKEN_SECRET, VSCodeSecretStorage } from './secret-storage'
+import path from 'path'
+
+let client: LanguageClient
+
+function sendCommandRequest(command: string, args: any[] | undefined): void {
+    client
+        .sendRequest('workspace/executeCommand', {
+            command,
+            arguments: args,
+        })
+        .catch(e => {
+            console.error(e)
+        })
+}
 
 export function activate(context: vscode.ExtensionContext): Promise<ExtensionApi> {
     console.log('Cody extension activated')
 
+    let serverOptions: ServerOptions = {
+        run: {
+            command: path.join(context.extensionPath, "resources", "bin", "llmsp"),
+            transport: TransportKind.stdio,
+        },
+        debug: {
+            command: path.join(context.extensionPath, "resources", "bin", "llmsp"),
+            transport: TransportKind.stdio,
+        },
+    }
+
     PromptMixin.add(languagePromptMixin(vscode.env.language))
 
+    vscode.commands.registerCommand
+
+    // Options to control the language client
+    let clientOptions: LanguageClientOptions = {
```

Comment:
this should be const IMO as we do not change the reference to clientOptions later on:
```suggestion
    const clientOptions: LanguageClientOptions = {
```